packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -621,28 +700,30 @@ contract OPContractsManagerUpgrader is OPContractsManagerBase {
     /// @notice Deploys and sets a new dispute game implementation
     /// @param _l2ChainId The L2 chain ID
     /// @param _disputeGame The current dispute game implementation
+    /// @param _newAnchorStateRegistryProxy The new anchor state registry proxy
     /// @param _gameType The type of game to deploy
     /// @param _opChainConfig The OP chain configuration
-    /// @param _blueprints The blueprint addresses
-    /// @param _implementations The implementation addresses
     /// @param _opChainAddrs The OP chain addresses
     function deployAndSetNewGameImpl(
         uint256 _l2ChainId,
         IDisputeGame _disputeGame,
+        IAnchorStateRegistry _newAnchorStateRegistryProxy,
         GameType _gameType,
         OPContractsManager.OpChainConfig memory _opChainConfig,
-        OPContractsManager.Blueprints memory _blueprints,
-        OPContractsManager.Implementations memory _implementations,
         ISystemConfig.Addresses memory _opChainAddrs
     )
         internal
     {
+        OPContractsManager.Blueprints memory bps = getBlueprints();
+        OPContractsManager.Implementations memory impls = getImplementations();
+
         // Get the constructor params for the game
         IFaultDisputeGame.GameConstructorParams memory params =
             getGameConstructorParams(IFaultDisputeGame(address(_disputeGame)));
 
         // Modify the params with the new vm values.
-        params.vm = IBigStepper(_implementations.mipsImpl);
+        params.anchorStateRegistry = IAnchorStateRegistry(address(_newAnchorStateRegistryProxy));
+        params.vm = IBigStepper(impls.mipsImpl);
         if (Claim.unwrap(_opChainConfig.absolutePrestate) == bytes32(0)) {
             revert OPContractsManager.PrestateNotSet();
         }
```

Comment:
This change will make it much easier to ship upgrades quickly. 

In cases where only the contracts are being upgraded, this would prevent us from having to look it up and resubmit it.


It would also avoid the situation we have with the upgrade 13 runbooks for sepolia, which are blocked because we don't yet know what the `absolutePrestate` is going to be at the time we do this upgrade. 


```suggestion
        if (Claim.unwrap(_opChainConfig.absolutePrestate) != bytes32(0)) {
            params.absolutePrestate = _opChainConfig.absolutePrestate;
        }
```

Reply:
How would this work? If it's unset, wouldn't the prestate be set to 0?

Reply:
No, it's subtle, but because the `params` value is taken from an existing game:
```
IFaultDisputeGame.GameConstructorParams memory params =
            getGameConstructorParams(IFaultDisputeGame(address(_disputeGame)));
```

it's already set, so this would just keep the old value.

I made a mock up in a draft PR on top of yours: https://github.com/ethereum-optimism/optimism/pull/14767

Reply:
Got it, I'll include this in the fast-follow PR

Reply:
Thanks.